playbooks/aws/openshift-cluster/build_ami.yml

```diff
@@ -6,64 +6,44 @@
   - name: get the necessary vars for ami building
     include_vars: vars.yml
 
-  - name: create a vpc with the name <clusterid>
+  - name: create an instance and prepare for ami
     include_role:
-      name: openshift_aws_vpc
+      name: openshift_aws
+      tasks_from: build_ami.yml
     vars:
-      r_openshift_aws_vpc_clusterid: "{{ provision.clusterid }}"
-      r_openshift_aws_vpc_cidr: "{{ provision.vpc.cidr }}"
-      r_openshift_aws_vpc_subnets: "{{ provision.vpc.subnets }}"
-      r_openshift_aws_vpc_region: "{{ provision.region }}"
-      r_openshift_aws_vpc_tags: "{{ provision.vpc.tags }}"
-      r_openshift_aws_vpc_name: "{{ provision.vpc.name | default(provision.clusterid) }}"
-
-  - name: create aws ssh keypair
-    include_role:
-      name: openshift_aws_ssh_keys
-    vars:
-      r_openshift_aws_ssh_keys_users: "{{ provision.instance_users }}"
-      r_openshift_aws_ssh_keys_region: "{{ provision.region }}"
-
-  - name: fetch the default subnet id
-    ec2_vpc_subnet_facts:
+      openshift_aws_clusterid: "{{ provision.clusterid }}"
```

Comment:
These variable assignments seem totally unnecessary.  They should live in defaults.yml of the role, or if we don't want users to modify these variables, in vars/ of the role.

Reply:
The clusterid is configurable.  Or maybe I missed something.  The default lives in defaults.yml and is `default`.  I'm overriding it here so that it gets set to to the value in `provision.clusterid`.

@michaelgugino, could you elaborate on this?